rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinTestUtils.java

```diff
@@ -125,4 +128,30 @@ public static byte[] flatKeysAsByteArray(List<BtcECKey> keys) {
 
         return flatPubKeys;
     }
+
+    public static void signTransactionInputFromMultiSigWithKeys(BtcTransaction transaction, int inputIndex, List<BtcECKey> keys) {
+        if (transaction.getWitness(inputIndex).getPushCount() == 0) {
+            signLegacyTransactionInputWithP2shMultiSigInputScript(transaction, inputIndex, keys);
+        }
+    }
+
+    private static void signLegacyTransactionInputWithP2shMultiSigInputScript(BtcTransaction transaction, int inputIndex, List<BtcECKey> keys) {
+        TransactionInput input = transaction.getInput(inputIndex);
+
+        Script inputRedeemScript = extractRedeemScriptFromInput(input)
+            .orElseThrow(() -> new IllegalArgumentException("Cannot sign inputs that are not from a multisig"));
+
+        Sha256Hash sigHash = transaction.hashForSignature(inputIndex, inputRedeemScript, BtcTransaction.SigHash.ALL, false);
+        Script inputScriptSig = input.getScriptSig();
+        for (BtcECKey key : keys) {
+            BtcECKey.ECDSASignature sig = key.sign(sigHash);
+            TransactionSignature txSig = new TransactionSignature(sig, BtcTransaction.SigHash.ALL, false);
+            byte[] txSigEncoded = txSig.encodeToBitcoin();
+
+            int keyIndex = inputScriptSig.getSigInsertionIndex(sigHash, key);
+            Script outputScript = createP2SHOutputScript(inputRedeemScript);
```

Comment:
This can probably be calculated only once, after getting the redeem script

Reply:
you right, thanks

Reply:
updated